### DISCUSSION TITLE
Delete unused/selected levels from menu/right click.

### DISCUSSION THREAD

**Discussion by F-Burning** [opentoonz/opentoonz#6517] [Ideas]
I think a way to delete unused or selected levels from xsheet with right click or in the menu above would be very useful. I don't referring clean levels from manager, but Browser's folders.

**Comment by RodneyBaker** [comment]
Not sure about Right Click menu but you could assign 'Remove Unused Levels' to a shortcut key.
(It's great fun creating levels, removing them and then watching them disappear via say... Alt X)  ;)

![image](https://user-images.githubusercontent.com/2576432/160258175-01b2d660-131e-4ede-b7a9-e47d4b6bc610.png)

**Comment by F-Burning** [comment]
I mean, I would like to delete levels in the Browser folders from Xsheet/top menu, instead of only clear the scene cast :P

**Comment by RodneyBaker** [comment]
In other words... delete level files from your harddrive?
If removed from the browser that is generally the case... the files referenced (by level or sequence) are no longer on the harddrive.**

I suppose what I'm trying to best understand is what is meant by the use of the term 'delete'.
Technically levels aren't deleted from the browser but rather are deleted from the harddrive.
The browser then in turn adjust it's listing to no longer display the file.
This in contrast to 'removing' a level from the Browser in which case the file may still be on the harddrive but just not appear in the Browser.
The xsheet (exposure sheet) then simply exposes drawings/levels without particular regard for whether they exist outside of the xsheet.  Their primary concern being to expose things that exist... levels... frames from levels...
If desiring a direct route from the exposure sheet to deletion of files on the harddrive we'd definitely want to have a warning pop up to allow confirmation from the user that the file they are about to delete will not be recoverable as it will be gone.

> I don't referring clean levels from manager, but Browser's folders.

This is perhaps the part I don't quite follow.
In the Browser we can Right Click a selection and choose Delete so I must assume you aren't referring to this.
You also don't seem to refer to levels in the xsheet as we can Right Click and Delete those exposures already.
It seems your desire is to vanquish both exposure and original file at same time.


**All of the above assumes the Preference option is on to keep the browser listing up to date by monitoring files on the harddrive.  If the option is turned off what is seen in the browser may not accurately reflect actual files on the harddrive. The files in this case may be removed from the listing but perhaps still present (i.e. not deleted) on the harddrive.

**Comment by RodneyBaker** [comment]
@F-Burning We could use more information regarding this request to lock it down.
On the surface the request seems pretty straightforward but we need to be very specific where it comes to deleting files and destructive processes in general.

Perhaps an option called "Destroy Level" would suffice as a clear enough warning (with secondary confirmation dialogue of course).

**Comment by RodneyBaker** [comment]
Propose an adjustment to make the Right Click menu customizable in (slightly) similar fashion as standard Opentoonz menus.  

I understand there is a difference in these context sensitive commands versus shortcut/menu commands and we need to preserve that but bridging the gap by allowing additional right click options would resolve the inevitable requests to add additional options. Today we might add this option to allow deleting unused/selected levels but tomorrow users will desire a different option.

Allowing the user to add options to the right click menu can resolve this.

One potential way forward would be to add a section at the bottom of the right click context menu where standard shortcuts can be added by the user as with standard menus.  In this way the context sensitive options would still be maintained at the top (uneditable except via source code) and the user customized additions would be at the botton.  Presumably an edit option for that section would be located here as well.
